system/Database/SQLite3/Forge.php

```diff
@@ -152,18 +151,42 @@ protected function _alterTable(string $alterType, string $table, $processedField
                     $fieldsToModify[] = $field;
                 }
 
-                (new Table($this->db, $this))
+                return (new Table($this->db, $this))
                     ->fromTable($table)
                     ->modifyColumn($fieldsToModify)
                     ->run();
 
-                return null; // Why null?
-
             default:
                 return parent::_alterTable($alterType, $table, $processedFields);
         }
     }
 
+    /**
+     * @param array|string $columnNames column names to DROP
+     *
+     * @return bool
+     *
+     * @throws DatabaseException
+     */
+    public function dropColumn(string $table, $columnNames)
+    {
+        $sqlExecuteResult = $this->_alterTable('DROP', $this->db->DBPrefix . $table, $columnNames);
```

Comment:
You should not call `$this->_alterTable()` in this method.

Reply:
I'm working on it.

Reply:
```
 ------ ---------------------------------------------------------------------------------------------------------------------------------- 
  Line   Forge.php                                                                                                                         
 ------ ---------------------------------------------------------------------------------------------------------------------------------- 
  :164   Method CodeIgniter\Database\SQLite3\Forge::dropColumn() has parameter $columnNames with no value type specified in iterable type  
         array.                                                                                                                            
         💡 See: https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type                                         
 ------ ----------------------------------------------------------------------------------------------------------------------------------
```

May I ask how to solve this error without modifying the parent doc type definition?

Reply:
```diff
diff --git a/phpstan-baseline.php b/phpstan-baseline.php
index 1b36c4a8f8..eb5bd75b5b 100644
--- a/phpstan-baseline.php
+++ b/phpstan-baseline.php
@@ -2683,12 +2683,6 @@ $ignoreErrors[] = [
    'count' => 1,
    'path' => __DIR__ . '/system/Database/Forge.php',
 ];
-$ignoreErrors[] = [
-   // identifier: missingType.iterableValue
-   'message' => '#^Method CodeIgniter\\\\Database\\\\Forge\\:\\:dropColumn\\(\\) has parameter \\$columnNames with no value type specified in iterable type array\\.$#',
-   'count' => 1,
-   'path' => __DIR__ . '/system/Database/Forge.php',
-];
 $ignoreErrors[] = [
    // identifier: missingType.iterableValue
    'message' => '#^Method CodeIgniter\\\\Database\\\\Forge\\:\\:modifyColumn\\(\\) has parameter \\$fields with no value type specified in iterable type array\\.$#',
diff --git a/system/Database/Forge.php b/system/Database/Forge.php
index 595cd6156b..b7f6743634 100644
--- a/system/Database/Forge.php
+++ b/system/Database/Forge.php
@@ -778,7 +778,7 @@ class Forge
     }
 
     /**
-     * @param array|string $columnNames column names to DROP
+     * @param list<string>|string $columnNames column names to DROP
      *
      * @return bool
      *
diff --git a/system/Database/SQLite3/Forge.php b/system/Database/SQLite3/Forge.php
index 1abb32be0d..c63aa6c40f 100644
--- a/system/Database/SQLite3/Forge.php
+++ b/system/Database/SQLite3/Forge.php
@@ -154,7 +154,7 @@ class Forge extends BaseForge
     }
 
     /**
-     * @param array|string $columnNames column names to DROP
+     * @param list<string>|string $columnNames column names to DROP
      *
      * @return bool
      *
```

Reply:
Done!